server/src/main/scala/com/cloudera/livy/server/interactive/InteractiveSession.scala

```diff
@@ -121,12 +121,18 @@ class InteractiveSession(
     }
     builderProperties.put(RSCConf.Entry.SESSION_KIND.key, kind.toString)
 
-    mergeConfList(livyJars(livyConf), LivyConf.SPARK_JARS)
+    val (sparkVersionStr, scalaVersionStr) = LivySparkUtils.sparkSubmitVersion(livyConf)
```

Comment:
Yes, the purpose here is to combine Spark/Scala with Session. Currently Spark/Scala is a global settings, we need to change them to be session based, that will support difference Sparks in one Livy.